interview-questions.md

```diff
@@ -8,62 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: The javaScript built in method .map() takes the value, index, and array as parameters. Value is always required while using map, while index and array are optional and only used if needed.
+
+  Researched answer: Map is a function that can iterate through an array and return a new array with updated values of the same length. It takes predefined parameters such as value, index, and array. The index and array are optional and only used when the programmer needs it.
 
-  Researched answer:
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: The difference between map and filter are that map functions do not change the original array, but instead iterate through the array and returns a new array of the same lengtht. The filter function also will not change the original array, as it will iterate through the array, but it will return ONLY the values that are true for the new array and the value that satisfy the past conditions.  
 
-  Researched answer:
+  Researched answer: Looking at the syllabus, I found that even though map and filter contain the same parameters (value, index, array), the outcomes are different. Map is a function that is going to be useful when the programmer wants to iterate through each item in an array and return a new array with updated values with the same length. Filter function, on the other hand, will lopp through an array and will make a decision to bring back only the group that satisfy the stated condition. 
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
-
-  Researched answer:
-
+  Your answer: I think the difference between a function and a method is the object. When a function belongs to an object it is called a method.  
+  
+  Researched answer: After researching this question, I found that a function executes a list of statements, while a method is a function that is applied to an object. A function can be a method if there is an object associated with it, but a method isn't the same as a function, as a function doesn't necessarily have an object associated with it. So functions and methods are both functions, but a method just implies that the function is a property of an object.       
 
 
 5. What is object destructuring?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Object destructuring creates special types of variables that expose a key inside of an object. 
 
+  Researched answer: Object destructuring in Javascript is a way of nominating a variable. The best thing about it is that it lets you take something like an array or an object and unpack their values into individual variables and manipulates object properties. It is very useful because the programmer can take the properties of an object and break them out into individual, recllable variables.
```

Comment:
This will be very useful as we start moving into react making code cleaner and more concise.